polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -1517,17 +1517,28 @@ ValueCategory MLIRScanner::VisitCallExpr(clang::CallExpr *expr) {
   }
 
   /// If the callee is part of the SYCL namespace, we may not want the
-  /// GetOrCreateMLIRFunction to add this FuncOp to the functionsToEmit dequeu,
+  /// GetOrCreateMLIRFunction to add this FuncOp to the functionsToEmit deque,
   /// since we will create it's equivalent with SYCL operations. Please note
   /// that we still generate some functions that we need for lowering some
   /// sycl op.  Therefore, in those case, we set ShouldEmit back to "true" by
   /// looking them up in our "registry" of supported functions.
-  auto ShouldEmit =
-      !mlirclang::isNamespaceSYCL(callee->getEnclosingNamespaceContext());
-  std::string name;
-  MLIRScanner::getMangledFuncName(name, callee, Glob.getCGM());
-  if (isSupportedFunctions(name))
+  bool isSyclFunc =
+      mlirclang::isNamespaceSYCL(callee->getEnclosingNamespaceContext());
+  bool ShouldEmit = !isSyclFunc;
```

Comment:
I think this is not fully correct, as we could have a SYCL device function in a different namespace (and host functions in the SYCL namespace). In #6847, we add logic to know whether the code we're generating is to be run in the host or in a device. We should use that mechanism instead.